pkg/alertmanageroperator/petset.go

```diff
@@ -0,0 +1,180 @@
+			Name: fmt.Sprintf("%s-db", p.Name),
+			VolumeSource: v1.VolumeSource{
+				EmptyDir: &v1.EmptyDirVolumeSource{},
+			},
+		})
+	} else {
+		pvc := v1.PersistentVolumeClaim{
+			ObjectMeta: v1.ObjectMeta{
+				Name: fmt.Sprintf("%s-db", p.Name),
+			},
+			Spec: v1.PersistentVolumeClaimSpec{
+				AccessModes: []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce},
+				Resources:   vc.Resources,
+				Selector:    vc.Selector,
+			},
+		}
+		if len(vc.Class) > 0 {
+			pvc.ObjectMeta.Annotations = map[string]string{
+				"volume.beta.kubernetes.io/storage-class": vc.Class,
+			}
+		}
+		petset.Spec.VolumeClaimTemplates = append(petset.Spec.VolumeClaimTemplates, pvc)
+	}
+
+	if old != nil {
+		petset.Annotations = old.Annotations
+	}
+	return petset
+}
+
+func makePetSetService(p *spec.Alertmanager) *v1.Service {
+	svc := &v1.Service{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "alertmanager",
+		},
+		Spec: v1.ServiceSpec{
+			ClusterIP: "None",
+			Ports: []v1.ServicePort{
+				{
+					Name:       "web",
+					Port:       9093,
+					TargetPort: intstr.FromInt(9093),
+					Protocol:   v1.ProtocolTCP,
+				},
+				{
+					Name:       "mesh",
+					Port:       6783,
+					TargetPort: intstr.FromInt(6783),
+					Protocol:   v1.ProtocolTCP,
+				},
+			},
+			Selector: map[string]string{
+				"app": "alertmanager",
+			},
+		},
+	}
+	return svc
+}
+
+func makePetSetSpec(ns, name, image, version string, replicas int32) v1alpha1.PetSetSpec {
+	commands := []string{
+		"/bin/alertmanager",
+		fmt.Sprintf("-config.file=%s", "/etc/prometheus/alertmanager.yaml"),
+		fmt.Sprintf("-web.listen-address=:%d", 9093),
+		fmt.Sprintf("-mesh.listen-address=:%d", 6783),
+		fmt.Sprintf("-storage.path=%s", "/tmp/alertmanager-data"),
+	}
+	for i := int32(0); i < replicas; i++ {
+		commands = append(commands, fmt.Sprintf("-mesh.peer=%s-%d.%s.%s.svc", name, i, "alertmanager", ns))
+	}
+
+	terminationGracePeriod := int64(0)
+	return v1alpha1.PetSetSpec{
+		ServiceName: "alertmanager",
+		Replicas:    &replicas,
+		Template: v1.PodTemplateSpec{
+			ObjectMeta: v1.ObjectMeta{
+				Labels: map[string]string{
+					"app":          "alertmanager",
+					"alertmanager": name,
+				},
+				Annotations: map[string]string{
+					"pod.alpha.kubernetes.io/initialized": "true",
+				},
+			},
+			Spec: v1.PodSpec{
+				TerminationGracePeriodSeconds: &terminationGracePeriod,
+				Containers: []v1.Container{
+					{
+						Command: commands,
+						Name:    name,
+						Image:   image,
+						Ports: []v1.ContainerPort{
+							{
+								Name:          "web",
+								ContainerPort: 9093,
+								Protocol:      v1.ProtocolTCP,
+							},
+							{
+								Name:          "mesh",
+								ContainerPort: 6783,
+								Protocol:      v1.ProtocolTCP,
+							},
+						},
+						VolumeMounts: []v1.VolumeMount{
+							{
+								Name:      "config-volume",
+								MountPath: "/etc/prometheus",
+							},
+						},
```

Comment:
We do have to mount the PVC specified above, right? With subpath as for Prometheus I'd suppose.